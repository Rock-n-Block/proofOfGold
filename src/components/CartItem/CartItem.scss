.c-item {
  background: #191919;
  padding: 10px 20px 10px 35px;
  border-radius: 12px;

  display: flex;
  & + & {
    margin-top: 10px;
    .c-item__elem span {
      display: none;
    }
  }
  &__wrapper {
    display: grid;
    grid-template-columns: 60px 110px;
    grid-column-gap: 20px;
  }
  &__box {
    display: grid;
    grid-template-columns: 400px 115px 150px 1fr;
    grid-column-gap: 20px;
    margin-left: 20px;
  }
  &__close {
    cursor: pointer;
    &-box {
      display: flex;
      z-index: 10;
      align-items: center;
    }
  }
  &__img {
    width: 100%;
  }
  &__elem {
    position: relative;
    padding-top: 30px;
    display: flex;
    align-items: center;
    height: 60px;
    span {
      position: absolute;
      top: -35px;
      color: #fff;
    }
  }
  &__name {
    font-size: 16px;
    line-height: 20px;
  }
  &__cost {
    font-family: MB;
  }
  .counter {
    padding: 0;
    background: transparent;
    border: none;
  }
}
@media (max-width: 1200px) {
  .c-item {
    display: flex;
    position: relative;
    &__wrapper {
      display: block;
      max-width: 30%;
    }
    &__close {
      &-box {
        position: absolute;
        top: 15px;
        right: 15px;
      }
    }
  }
}

@media (max-width: 1024px) {
  .c-item {
    padding: 20px;
    & + & {
      margin-top: 30px;
      .c-item__elem span {
        display: block;
      }
    }
    &__box {
      display: flex;
      flex-direction: column;
    }
    &__elem {
      height: auto;
      padding-top: 0;
      flex-direction: column;
      align-items: flex-start;
      & + & {
        margin-top: 50px;
      }
      span {
        position: static;
        margin-bottom: 10px;
      }
    }
  }
}

@media (max-width: 767px) {
  .c-item {
    &__elem {
      span {
        font-size: 16px;
        line-height: 20px;
      }
    }
  }
}
