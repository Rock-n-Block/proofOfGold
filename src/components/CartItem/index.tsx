import React from 'react';
import { observer } from 'mobx-react-lite';
import { Link } from 'react-router-dom';

import { Counter } from '../../components';
import numberWithCommas from '../../utils/numberWithCommas';
import { useMst } from '../../store/root';

import './CartItem.scss';

import deleteImg from '../../assets/img/cart/delete.svg';

export interface CartItemProps {
  image: string;
  id: string;
  name: string;
  price: number;
  quantity: number;
  isFirst?: boolean;
  supply: number;
  totalPrise: number;
  addMore: () => void;
  deleteOneOf: () => void;
  deleteCurrent: () => void;
  total_supply: number;
}

const CartItem: React.FC<CartItemProps> = observer(
  ({
    id,
    image,
    name,
    price,
    quantity,
    totalPrise,
    addMore,
    deleteOneOf,
    supply,
    deleteCurrent,
  }) => {
    return (
      <div className="c-item">
        <div className="c-item__wrapper">
          <div className="c-item__close-box">
            <img
              src={deleteImg}
              className="c-item__close"
              onClick={() => deleteCurrent()}
            />
          </div>
          <Link to={`/product/${id}`}>
            <img src={`https://${image}`} alt="" className="c-item__img" />
          </Link>
        </div>
        <div className="c-item__box">
          <div className="c-item__elem">
            <span className="text-bold">PRODUCT</span>
            <div className="c-item__name">{name}</div>
          </div>
          <div className="c-item__elem">
            <span>PRICE</span>
            <div className=" text-lg text-gradient c-item__cost">
              ${numberWithCommas(price)}
            </div>
          </div>
          <div className="c-item__elem">
            <span>QUANTITY</span>
            <Counter
              onIncrease={addMore}
              onDecrease={deleteOneOf}
              min={0}
              max={supply}
              isInput={false}
              value={quantity}
              onChange={() => {}}
            />
          </div>
          <div className="c-item__elem">
            <span>SUBTOTAL</span>
            <div className=" text-lg text-gradient c-item__cost">
              ${numberWithCommas(totalPrise)}
            </div>
          </div>
        </div>
      </div>
    );
  },
);

export default CartItem;
