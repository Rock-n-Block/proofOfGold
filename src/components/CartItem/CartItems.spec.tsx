import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import CartItem from './index';
import { CartItemProps } from './index';

const cartItemData = {
  bonus_coins: 32.5,
  quantity: 1,
  description: null,
  group: 'gold_coin',
  id: '4',
  image: 'd-pog.com/media/1614960293.494851.png',
  lucky_prize: 400,
  name: '21 grams â€“ Ducatus Prime Gold Coin',
  price: 1877.21,
  reviews: [],
  sold: 0,
  supply: 1597,
  total_supply: 1597,
  totalPrise: 1877.21,
  addMore: () => {},
  deleteOneOf: () => {},
  deleteCurrent: () => {},
};

const setUp = (props?: CartItemProps): ShallowWrapper =>
  shallow(<CartItem {...cartItemData} {...props} />);

describe('should render CartItem component', () => {
  let component: ShallowWrapper;
  beforeEach(() => {
    component = setUp();
  });
  it('should CartItem Render', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should deleteCurrent deleted', () => {
    const btn = component.find('.c-item__close');
    btn.simulate('click');

    expect(component).toMatchSnapshot();
  });
});
