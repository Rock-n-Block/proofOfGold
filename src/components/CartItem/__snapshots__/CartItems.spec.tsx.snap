// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartItem component Should deleteCurrent deleted 1`] = `
<div
  className="c-item"
>
  <div
    className="c-item__wrapper"
  >
    <div
      className="c-item__close-box"
    >
      <img
        className="c-item__close"
        onClick={[Function]}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      />
    </div>
    <Link
      to="/product/4"
    >
      <img
        alt=""
        className="c-item__img"
        src="https://d-pog.com/media/1614960293.494851.png"
      />
    </Link>
  </div>
  <div
    className="c-item__box"
  >
    <div
      className="c-item__elem"
    >
      <span
        className="text-bold"
      >
        PRODUCT
      </span>
      <div
        className="c-item__name"
      >
        21 grams – Ducatus Prime Gold Coin
      </div>
    </div>
    <div
      className="c-item__elem"
    >
      <span>
        PRICE
      </span>
      <div
        className=" text-lg text-gradient c-item__cost"
      >
        $
        1,877.21
      </div>
    </div>
    <div
      className="c-item__elem"
    >
      <span>
        QUANTITY
      </span>
      <Memo(Counter)
        isInput={false}
        max={1597}
        min={0}
        onChange={[Function]}
        onDecrease={[Function]}
        onIncrease={[Function]}
        value={1}
      />
    </div>
    <div
      className="c-item__elem"
    >
      <span>
        SUBTOTAL
      </span>
      <div
        className=" text-lg text-gradient c-item__cost"
      >
        $
        1,877.21
      </div>
    </div>
  </div>
</div>
`;

exports[`should render CartItem component should CartItem Render 1`] = `
<div
  className="c-item"
>
  <div
    className="c-item__wrapper"
  >
    <div
      className="c-item__close-box"
    >
      <img
        className="c-item__close"
        onClick={[Function]}
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      />
    </div>
    <Link
      to="/product/4"
    >
      <img
        alt=""
        className="c-item__img"
        src="https://d-pog.com/media/1614960293.494851.png"
      />
    </Link>
  </div>
  <div
    className="c-item__box"
  >
    <div
      className="c-item__elem"
    >
      <span
        className="text-bold"
      >
        PRODUCT
      </span>
      <div
        className="c-item__name"
      >
        21 grams – Ducatus Prime Gold Coin
      </div>
    </div>
    <div
      className="c-item__elem"
    >
      <span>
        PRICE
      </span>
      <div
        className=" text-lg text-gradient c-item__cost"
      >
        $
        1,877.21
      </div>
    </div>
    <div
      className="c-item__elem"
    >
      <span>
        QUANTITY
      </span>
      <Memo(Counter)
        isInput={false}
        max={1597}
        min={0}
        onChange={[Function]}
        onDecrease={[Function]}
        onIncrease={[Function]}
        value={1}
      />
    </div>
    <div
      className="c-item__elem"
    >
      <span>
        SUBTOTAL
      </span>
      <div
        className=" text-lg text-gradient c-item__cost"
      >
        $
        1,877.21
      </div>
    </div>
  </div>
</div>
`;
